build-logic/src/main/kotlin/dokkabuild.test-k2.gradle.kts

```diff
@@ -9,39 +12,81 @@
 plugins {
     id("dokkabuild.base")
     id("dokkabuild.java")
+    `jvm-test-suite`
 }
 
-val descriptorsTestConfiguration: Configuration by configurations.creating {
-    extendsFrom(configurations.testImplementation.get())
-}
-val symbolsTestConfiguration: Configuration by configurations.creating {
-    extendsFrom(configurations.testImplementation.get())
+val symbolsTestImplementation: Configuration by configurations.creating {
+    description = "Dependencies for symbols tests (K2)"
+    declarable()
 }
 
-val symbolsTest = tasks.register<Test>("symbolsTest") {
-    useJUnitPlatform {
-        excludeTags("onlyDescriptors", "onlyDescriptorsMPP")
-    }
-    classpath += symbolsTestConfiguration
-}
-// run symbols and descriptors tests
-tasks.test {
-    //enabled = false
-    useJUnitPlatform {
-        excludeTags("onlySymbols")
-    }
-    classpath += descriptorsTestConfiguration
-    dependsOn(symbolsTest)
+val symbolsTestImplementationResolver: Configuration by configurations.creating {
+    description = "Resolve dependencies for symbols tests (K2)"
+    resolvable()
+    extendsFrom(symbolsTestImplementation)
+    attributes { jvmJar(objects) }
 }
 
-val descriptorsTest = tasks.register<Test>("descriptorsTest") {
-    useJUnitPlatform {
-        excludeTags("onlySymbols")
-    }
-    classpath += descriptorsTestConfiguration
+val descriptorsTestImplementation: Configuration by configurations.creating {
+    description = "Dependencies for descriptors tests (K1)"
+    declarable()
 }
 
-tasks.check {
-    dependsOn(symbolsTest)
+val descriptorsTestImplementationResolver: Configuration by configurations.creating {
+    description = "Resolve dependencies for descriptors tests (K1)"
+    resolvable()
+    extendsFrom(descriptorsTestImplementation)
+    attributes { jvmJar(objects) }
 }
 
+@Suppress("UnstableApiUsage")
+testing {
+    suites {
+        val test by suites.getting(JvmTestSuite::class) {
+
+            // JUnit tags for descriptors (K1) and symbols (K2) are defined with annotations in test classes.
+            val onlyDescriptorTags = listOf("onlyDescriptors", "onlyDescriptorsMPP")
+            val onlySymbolsTags = listOf("onlySymbols")
+
+            // Modify the regular :test target to exclude symbols (K2) tests.
+            targets.named("test") {
+                testTask.configure {
+                    description = "Runs tests using descriptors-analysis (K1) (excluding tags: $onlySymbolsTags)"
+                    useJUnitPlatform {
+                        excludeTags.addAll(onlySymbolsTags)
+                    }
+                    classpath += descriptorsTestImplementationResolver.incoming.files
+                }
+            }
+
+            // Create a new target for _only_ running descriptor (K1) tests.
+            targets.register("descriptorsTest") {
```

Comment:
BTW, after some thoughts, may be we need to rename tasks to `testSymbols` and `testDescriptors` - it will be more aligned with integration tests naming (like `testTemplateProject***`) and also tasks will be near each other in IDEA?

Reply:
I'd be happy to rename them! Your suggestions make sense to me.

I wanted to keep the names the same, so that this PR didn't disrupt anyone's habits.

Reply:
Just let me know what you prefer, otherwise I'll leave them as-is in this PR. We can always change them later.

Reply:
For me it's fine to change it now. @vmishenev WDYT?

Reply:
I think so, `testSymbols` and `testDescriptors` are better

Reply:
Done!